src/daft-plan/src/logical_plan.rs

```diff
@@ -46,6 +47,79 @@ impl LogicalPlan {
         }
     }
 
+    pub fn required_columns(&self) -> Vec<HashSet<String>> {
+        match self {
+            Self::Limit(..) | Self::Coalesce(..) => vec![HashSet::new()],
```

Comment:
Potential future efficiency optimization: instead of returning a `Vec<HashSet<String>>`, have `required_columns` take a single `&mut HashSet<String>` that this method adds to, and do the same for `get_required_columns()` at the `Expr` level. This way, we won't have any intermediate `Vec` or `HashSet` allocations that will just be consolidated into a single `HashSet` later.